Microsoft.Research/Contracts/System.Data/System.Data.DataRowCollection.cs

```diff
@@ -0,0 +1,220 @@
+﻿// CodeContracts
+// 
+// Copyright (c) Microsoft Corporation
+// 
+// All rights reserved. 
+// 
+// MIT License
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections;
+
+namespace System.Data
+{
+    using System.Diagnostics.Contracts;
+
+    /// <summary>
+  /// Represents a collection of rows for a <see cref="T:System.Data.DataTable"/>.
+  /// </summary>
+  /// <filterpriority>1</filterpriority>
+  public sealed class DataRowCollection // : InternalDataCollectionBase
+  {
+    /// <summary>
+    /// Gets the total number of <see cref="T:System.Data.DataRow"/> objects in this collection.
+    /// </summary>
+    /// 
+    /// <returns>
+    /// The total number of <see cref="T:System.Data.DataRow"/> objects in this collection.
+    /// </returns>
+    //public override int Count
+    //{
+    //  get
+    //  {
+    //    return 0;
+    //  }
+    //}
+
+    /// <summary>
+    /// Gets the row at the specified index.
+    /// </summary>
+    /// 
+    /// <returns>
+    /// The specified <see cref="T:System.Data.DataRow"/>.
+    /// </returns>
+    /// <param name="index">The zero-based index of the row to return. </param><exception cref="T:System.IndexOutOfRangeException">The index value is greater than the number of items in the collection. </exception><filterpriority>1</filterpriority><PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" PathDiscovery="*AllFiles*"/></PermissionSet>
```

Comment:
:bulb: The formatting here (and in places that follow in the same file) ... "could be improved". :wink: